test/integration_test/basic_dmthin_test.go

```diff
@@ -22,124 +22,259 @@ import (
 )
 
 var (
-	pxVer2_13_0, _ = version.NewVersion("2.13.0")
+	pxVer3_0_0, _ = version.NewVersion("3.0.0")
+)
+
+type TestOptionsPresent int
+
+const (
+	PxStoreV2Present TestOptionsPresent = iota
+	MetadataPresent
+	AutoJournalPresent
+	KvdbDevicePresent
+	JournalPresent
+	MaxOptions
 )
 
-func defaultDmthinSpec(t *testing.T) *corev1.StorageCluster {
+type OptionsArr [MaxOptions]bool
+
+func defaultDmthinSpec(t *testing.T, addDmthinOption bool) *corev1.StorageCluster {
```

Comment:
nit.. portworx.io/misc-args is necessary for dmthin. In the absence of this param this function will not return a dmthin spec. Right?
In that case we should rename this function and maybe call it defaultSpec only.